code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -410,14 +421,68 @@
 	else
 		to_chat(usr, "<span class='warning'>This mob type can't use this verb.</span>")
 
-/obj/structure/closet/update_icon()//Putting the welded stuff in update_icon() so it's easy to overwrite for special cases (Fridges, cabinets, and whatnot)
+/obj/structure/closet/update_icon()
 	cut_overlays()
 	if(!opened)
-		icon_state = icon_closed
-		if(welded)
-			add_overlay(welded_overlay_state)
+		layer = OBJ_LAYER
+		if(!is_animating_door)
+			if(icon_door)
+				add_overlay("[icon_door]_door")
+			else
+				add_overlay("[icon_state]_door")
+			if(secure && !broken)
+				if(locked)
+					add_overlay("locked")
+				else
+					add_overlay("unlocked")
+			if(welded)
+				add_overlay(icon_welded)
+
 	else
-		icon_state = icon_opened
+		layer = BELOW_OBJ_LAYER
+		if(!is_animating_door)
+			if(icon_door_override)
+				add_overlay("[icon_door]_open")
+			else
+				add_overlay("[icon_state]_open")
+
+/obj/structure/closet/proc/animate_door(var/closing = FALSE)
+	if(!door_anim_time)
```

Comment:
Well, it's not really needed since the animation parameters are handpicked rather than being generated on the fly. Although it might come in handy in the future